### DISCUSSION TITLE
Colour palette?

### DISCUSSION THREAD

**Discussion by dev-zero** [wezterm/wezterm#1290] [Q&A]
Hi there,

I am using Wezterm with the OneHalfDark color scheme. One of the apps I use explicitly uses a black font color for some of its output when colorizing. Those parts of the output are not visible. The bug report for that app which shows the problem is here: https://github.com/spack/spack/issues/11116
One commenter says: "This seems to be a Terminal, Palette issue". Is there something I could change in my Wezterm config (instead of changing the theme) to fix this?

Thanks in advance,
Tiziano

**Comment by wez** [comment]
The issue is with the OneHalfDark theme; to resolve it, the theme needs to be changed.
The themes come from this upstream project: https://github.com/mbadolato/iTerm2-Color-Schemes

To make progress locally, you can make your own version of that theme by:

 ```
mkdir -p ~/.config/wezterm/colors
curl -o ~/.config/wezterm/colors/OneHalfDark.toml https://raw.githubusercontent.com/wez/wezterm/main/assets/colors/OneHalfDark.toml
```

Then you can edit the offending color, and perhaps propose that change to the upstream project as a follow up!

See also https://wezfurlong.org/wezterm/config/appearance.html#defining-a-color-scheme-in-a-separate-file

**Reply by dev-zero** [reply]
Thanks for the quick answer.
While looking around for other schemes I've discovered the [base16](https://github.com/chriskempson/base16) project. Here's my attempt at providing a template for wezterm: https://github.com/dev-zero/base16-wezterm/blob/main/templates/default.mustache, probably needs some tweaking still.
Not sure yet how to extend it to 256 colors, see the example for the kitty template: https://github.com/kdrag0n/base16-kitty/blob/master/templates/default-256.mustache

**Comment by dev-zero** [comment]
@wez sorry for bothering you again, but a direct comparison of the same color scheme with kitty seems to indicate that it's not just the color scheme:

![20211115_08h45m25s_grim](https://user-images.githubusercontent.com/11307/141742027-6a5900d1-35d4-4606-8571-99ca26f09355.png)

This is with the original color scheme you ship, as well as with one where I adjusted the `selection_bg/_fg` to match the one from kitty.

Here's the kitty config:

```
# Onehalf Colorscheme for Kitty
# Based on https://github.com/sonph/onehalf
# By https://github.com/dbinary

foreground            #dcdfe4
background            #282c34
selection_foreground  #000000
selection_background  #FFFACD
url_color             #0087BD

# black
color0   #282c34
color8   #5d677a

# red
color1   #e06c75
color9   #e06c75

# green
color2   #98c379
color10  #98c379

# yellow
color3   #e5c07b
color11  #e5c07b

# blue
color4  #61afef
color12 #61afef

# magenta
color5   #c678dd
color13  #c678dd

# cyan
color6   #56b6c2
color14  #56b6c2

# white
color7   #dcdfe4
color15  #dcdfe4
```

and here my wezterm color scheme:

```
# OneHalfDark
[colors]
foreground = "#dcdfe4"
background = "#282c34"
cursor_bg = "#a3b3cc"
cursor_border = "#a3b3cc"
cursor_fg = "#dcdfe4"
#selection_bg = "#474e5d"
#selection_fg = "#dcdfe4"
selection_bg = "#fffacd"
selection_fg = "#000000"

ansi = ["#282c34","#e06c75","#98c379","#e5c07b","#61afef","#c678dd","#56b6c2","#dcdfe4"]
brights = ["#282c34","#e06c75","#98c379","#e5c07b","#61afef","#c678dd","#56b6c2","#dcdfe4"]
```

**Reply by wez** [reply]
I'm sorry, I can't tell what's up; can you explain the issue?

**Reply by dev-zero** [reply]
Sure :) on the left side is kitty, on the right side is wezterm. Both with the same colour scheme (OneHalf Dark, see config above). On the kitty side are the dashes visible on the left (in the first character column) of the tree, while with wezterm they have the same color as the background (when selecting text they become visbile), even though the background of wezterm is darker than kitty (despite having the same hex value). Why is that?
@wez

**Reply by dev-zero** [reply]
Same story with `iTerm2`: there the dashes are also visible with the same color scheme.

**Reply by dev-zero** [reply]
@wez sorry to bother you again, the issue is basically that an iTerm2 scheme which is usable (e.g. everything is legible) with iTerm2 or Kitty results in foreground text to be the same as background when loaded in wezterm

**Reply by wez** [reply]
I don't have enough information to say what's happening.

I suspect that https://wezfurlong.org/wezterm/config/lua/config/bold_brightens_ansi_colors.html might be relevant.  Kitty doesn't have this option, and is unusual in that its behavior matches `bold_brightens_ansi_colors = false`.  (Trivia: that's one of the reasons I created wezterm).

Please file an issue that includes your configuration and capture and attach a session recording of the offending screen.  You can use https://github.com/wez/wezterm/blob/main/wt-record to do that, or manually run the `script` utility.  Please keep the recorded terminal at the same fixed 80x24 size throughout.
